src/mlpack/tests/main_tests/hmm_train_test.cpp

```diff
@@ -51,6 +51,70 @@ inline void FileExists(std::string fileName)
   ifp.close();
 }
 
+inline bool ApproximatelyEqual(HMMModel& h1,
+                               HMMModel& h2,
+                               double tolerance)
+{
+  if (h1.Type() != h2.Type())
+    return false;
+  HMMType  hmmType = h1.Type();
+  bool transitionEqual = false;
+  bool emissionEqual = false;
+  bool initialEqual = false;
+  if (hmmType ==  DiscreteHMM)
+  {
+    transitionEqual = approx_equal(
+        h1.DiscreteHMM()->Transition(),
+        h2.DiscreteHMM()->Transition(),
+        "absdiff",
+        tolerance
+        );
+    initialEqual = approx_equal(
+        h1.DiscreteHMM()->Transition(),
+        h2.DiscreteHMM()->Transition(),
+        "absdiff",
+        tolerance
+        );
+    // TODO
+    emissionEqual = true;
```

Comment:
Don't forget the todo here. :)  (I am writing this comment in part so that I remember to double-check that this gets done before merge... probably you already know that you still need to do it. :))